ols/app/models/models.py

```diff
@@ -135,6 +135,18 @@ class ReferencedDocument:
     docs_url: str
     title: str
 
+    @staticmethod
+    def from_rag_chunks(rag_chunks: list["RagChunk"]) -> list["ReferencedDocument"]:
+        """Create a list of ReferencedDocument from a list of rag_chunks."""
+        return list(
+            {
+                rag_chunk.doc_url: ReferencedDocument(
```

Comment:
it is in original code, I know. But why we create dict just to convert it to list in the next step?

Reply:
To drop duplicates.

Reply:
`list(set(rag_chunks))`

Reply:
Sure, but ReferencedDocument needs to be hashable and I need to modify the other test to not relly on the order.
I can do that, no problem.

Reply:
updated

Reply:
While dict kinda retains the order of keys as they are inserted, if it is crucial to retain the same order as is in the rag_chunks, I would do this then.

```python
list(
    OrderedDict(
        (rag_chunk.doc_url, ReferencedDocument(rag_chunk.doc_url, rag_chunk.doc_title))
        for rag_chunk in summarizer_response.rag_chunks
    ).values()
)
```

Reply:
updated